### PR TITLE
docs(writing-guidelines): DDT-554 ux content writing guidelines v2

### PR DESCRIPTION
**Pull request by hynes-dialpad**
# UX Content Writing Guidelines v2

Person dancing in an odd chicken costume

- :hammer_and_wrench: Change Type: Documentation
- :book: Jira Ticket: [DDT-554](https://dialpad.atlassian.net/browse/DDT-554)

## :book: Description

This PR represents a major update to the current UX writing guidelines. While there are a number of minor updates to previous guidance, such as when to use sentence casing (previously it was always, now it's always except in actionable areas such as buttons, menus, etc), extensive documentation has been added to expand on how to write:

- Actionable language
- Help content
- Error messages
- Inclusive language 

Grammar and mechanics as well as voice and tone areas were broken out into their own areas.

## :bulb: Context

These changes are being made to better help product designers write and utilize clear, concise copy writing in their work as well as making sure the overall voice, tone, and approach is consistent.

## :pencil: Checklist

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

[DDT-554]: https://dialpad.atlassian.net/browse/DDT-554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/dialtone-documentation/docs/guides/content/grammar-and-mechanics.md
Comment (francisrupert) [inline apps/dialtone-documentation/docs/guides/content/grammar-and-mechanics.md]: We already have a VuePress doc component for "Do/Don't".

https://github.com/dialpad/dialtone/blob/staging/apps/dialtone-documentation/docs/components/button.md?plain=1#L48-L61

Comment (hynes-dialpad) [reply apps/dialtone-documentation/docs/guides/content/grammar-and-mechanics.md]: Ok. I'll update them. I'll probably update the Do/Don't component as well.

**Comment by francisrupert** [comment]
> Move Content into it's own area from Guides. Still need a TOC overview page.

@juliodialpad is working on enabling the nested left nav hierarchy so this can remain within **Guides** section. Or... would you prefer "Content" to indeed be a top-level nav item and *not* within "Guides"?

**Comment by juliodialpad** [comment]
Opened #406 that was based on this branch and will be merged back, please review @francisrupert @hynes-dialpad

**Comment by hynes-dialpad** [comment]
We can merge Julio's PR here. Maybe we move Content out eventually but for now let's keep it as-is. Thanks!

**Comment by francisrupert** [comment]
> We can merge Julio's PR here.

Yeah, that's his intent. https://github.com/dialpad/dialtone/pull/406 gets the functionality working, then he'll merge that branch onto your (#352).

File: apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue
Comment (juliodialpad) [inline apps/dialtone-documentation/docs/.vuepress/theme/components/Home.vue]:
```suggestion
          <router-link class="d-link" to="/guides/content/">
```

File: apps/dialtone-documentation/docs/_data/site-nav.json
Comment (juliodialpad) [inline apps/dialtone-documentation/docs/_data/site-nav.json:99]: Don't think Assets need to be capitalized.

```suggestion
          "text": "Design assets",
```

**Review by juliodialpad — APPROVED** [review]
This looks of from the technical point of view, defer to @francisrupert @fede-dp to check that the wording and styling looks correct.

Just apply the small fix suggested.

File: apps/dialtone-documentation/docs/_data/site-nav.json
Comment (hynes-dialpad) [reply apps/dialtone-documentation/docs/_data/site-nav.json:99]: Per the new writing guidance, menu labels should be title-cased

Comment (juliodialpad) [reply apps/dialtone-documentation/docs/_data/site-nav.json:99]: We need to change all the titles then, we were not capitalizing previously, doesn't need to be in this PR, but we should create a ticket to not forgot about it

Comment (hynes-dialpad) [reply apps/dialtone-documentation/docs/_data/site-nav.json:99]: We might as well scrub the current doc site to correct this. I'll do it. Also for action items such as buttons.

**Review by francisrupert — COMMENTED** [review]
This is more @juliodialpad:

The nested left-nav is collapsed when directly linking to it, or even refreshing it. 

For example, directly linking to https://dialtone.dialpad.com/deploy-previews/pr-352/guides/content/help-content/.

**Comment by braddialpad** [comment]
> This is more @juliodialpad:
> 
> The nested left-nav is collapsed when directly linking to it, or even refreshing it.
> 
> For example, directly linking to https://dialtone.dialpad.com/deploy-previews/pr-352/guides/content/help-content/.

Definitely a problem but doesn't need to block this PR.

**Review by braddialpad — APPROVED** [review]
Didn't really review the content or grammar but looks good from a technical perspective.

**Review by francisrupert — APPROVED** [review]
Can you place source Figma images as a file in [Dialpad Design Website](https://www.figma.com/files/903342056393763336/project/17693047/Dialpad-Design-Website-(dialpad.design)?fuid=1093938355411584084).

**Comment by hynes-dialpad** [comment]
> Can you place source Figma images as a file in [Dialpad Design Website](https://www.figma.com/files/903342056393763336/project/17693047/Dialpad-Design-Website-(dialpad.design)?fuid=1093938355411584084).

Added!

**Pull request merged**
